binary_search_tree/binary_search_tree.py

```diff
@@ -1,17 +1,53 @@
 class BinarySearchTree:
-  def __init__(self, value):
-    self.value = value
-    self.left = None
-    self.right = None
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
 
-  def insert(self, value):
-    pass
+    def insert(self, value):
+        def go(value, curr):
+            if value > curr.value:
+                if curr.right is None:
+                    curr.right = BinarySearchTree(value)
+                    return
+                else:
+                    go(value, curr.right)
+            elif value < curr.value:
+                if curr.left is None:
+                    curr.left = BinarySearchTree(value)
+                    return
+                else:
+                    go(value, curr.left)
+        go(value, self)
 
-  def contains(self, target):
-    pass
+    def contains(self, target):
+        def go(value, curr):
+            if value == curr.value:
+                return True
+            elif value > curr.value:
+                if curr.right is None:
+                    return False
+                else:
+                    return go(value, curr.right)
+            elif value < curr.value:
+                if curr.left is None:
+                    return False
+                else:
+                    return go(value, curr.left)
+        return go(target, self)
 
```

Comment:
again, very nice solution but probably could have done it without the nested method.

Reply:
would include an else just to catch any edge cases that are not obvious right now